webapp/utils/channel_utils.jsx

```diff
@@ -190,40 +190,44 @@ export function showManagementOptions(channel, isAdmin, isSystemAdmin, isChannel
     return true;
 }
 
-export function showDeleteOption(channel, isAdmin, isSystemAdmin, isChannelAdmin, userCount) {
+export function showDeleteOptionForCurrentUser(channel) {
```

Comment:
This is the actual work. I got rid of the dubiously defined parameters and opted to fetch them myself inside of this function so that we always know what they mean. I also removed the "last person in a channel" loophole, and changed the authorization checks to handle the OR cases correctly.